src/models/model.cpp

```diff
@@ -463,8 +472,11 @@ void Model::CreateSessionOptionsFromConfig(const Config::SessionOptions& config_
       for (auto& option : provider_options.options) {
         opts.emplace(option.first, option.second);
       }
-
       session_options.AppendExecutionProvider("QNN", opts);
+    } else if (provider_options.name == "web") {
```

Comment:
nit: can we use "webgpu"? using 'web' as the provider in a mobile scenario would be somewhat confusing

Reply:
I was wondering too if we should use webgpu.
'web' is burned into model builder when you use '-e web' in model builder to create models for ort-web (fp16/int4 with MHA). It is stored in genai_config.json. But ort-web doesn't use genai so it does not look at that config file and we could change it.
But for most llm's there are models with that config file on huggingface for ort-web and we can just use them with genai if we look for 'web'.
Maybe look for 'web' and 'webgpu' and change model builder to insert 'webgpu' going forward ?

But there will be a new issue soon:
users would love to use the same model that are there for directml - working on that.
But their config file has 'directml' in it so we would need to be able to override this via api.

Reply:
Sounds like 'web' is more a preset (selects attention type and data sizes) than an EP selector and perhaps isn't 1:1 with any EP info that gets baked into genai_config.json. And I assume once you add GQA support the model builder's notion of what 'web' is will change accordingly and now just be a data type selector.

Maybe the genai config should have a set of preferred EPs and their settings. User selects an EP from that list at runtime. If there's only one EP (current state) the selection is automatic.

Reply:
yes, something like "would work on these ep's"

Reply:
> But ort-web doesn't use genai so it does not look at that config file and we could change it.
> Maybe look for 'web' and 'webgpu' and change model builder to insert 'webgpu' going forward ?

Since the model builder uses the execution provider flag to decide more specific information, I think we should replace `web` with `webgpu` to be more precise. Since ONNX Runtime GenAI isn't currently used for ONNX Runtime Web, I think it is fine to update any published `genai_config.json` files so that they use `webgpu` instead of `web`.

Reply:
> But ort-web doesn't use genai so it does not look at that config file and we could change it.
> Maybe look for 'web' and 'webgpu' and change model builder to insert 'webgpu' going forward ?

Since the model builder uses the execution provider flag to decide more specific information, I think we should replace `web` with the execution provider's name used in ONNX Runtime to be more precise. The execution provider's options are set [here](https://github.com/microsoft/onnxruntime-genai/blob/dfbe14c39bc0486e1289332bca2003ff66a74fc7/src/python/py/models/builder.py#L69) based on the name passed. Since ONNX Runtime GenAI isn't currently used for ONNX Runtime Web, I think it is fine to update any published `genai_config.json` files so that they use a different name instead of `web`.

Reply:
> But ort-web doesn't use genai so it does not look at that config file and we could change it.
> Maybe look for 'web' and 'webgpu' and change model builder to insert 'webgpu' going forward ?

Since the model builder uses the execution provider flag to decide more specific information, I think we should replace `web` with the execution provider's name used in ONNX Runtime to be more precise. The execution provider's options are set [here](https://github.com/microsoft/onnxruntime-genai/blob/dfbe14c39bc0486e1289332bca2003ff66a74fc7/src/python/py/models/builder.py#L69) based on the name passed. Since ONNX Runtime GenAI isn't currently used for ONNX Runtime Web, I think it is fine to update any published `genai_config.json` files so that they use a different name instead of `web`.

> users would love to use the same model that are there for directml - working on that.
> But their config file has 'directml' in it so we would need to be able to override this via api.
> Maybe the genai config should have a set of preferred EPs and their settings. User selects an EP from that list at runtime. If there's only one EP (current state) the selection is automatic.

An ordered list of execution providers with their settings can be passed in the `genai_config.json` [here](https://github.com/microsoft/onnxruntime-genai/blob/dfbe14c39bc0486e1289332bca2003ff66a74fc7/src/python/py/models/builder.py#L308). ONNX Runtime GenAI will process the execution providers in the specified order and move to the next execution provider in the list if the previous one fails just like ONNX Runtime does.